src/diagnostics_component/utils/file_info_utils/src/file_info_utils.c

```diff
@@ -224,7 +224,7 @@ bool FileInfoUtils_GetNewestFilesInDirUnderSize(
     }
 
     int fileIndex = 0;
-    unsigned long currentFileMaxCount = 0;
+    long currentFileMaxCount = 0;
```

Comment:
Why signed long? Instead, should this be unsigned int to match type of maxFileSize?
Also, line 235 should protect against overflow

Reply:
the type is tied to the long long type returned by stat. I'm going to change the casted type here to long long in order to validate what we've already seen